src/components/ComponentDef.js

```diff
@@ -37,13 +37,7 @@ ComponentDef.prototype = {
      * so that we can instantiate nested components before their parents.
      */
     ___addChild: function (componentDef) {
-        var children = this.___children;
-
-        if (children) {
-            children.push(componentDef);
-        } else {
-            this.___children = [componentDef];
-        }
+        (this.___children || (this.___children = [])).push(componentDef);
```

Comment:
This code style is followed the existing code. Sorry, I should have had more carefulness for the performance to contribute code.